uc_mcore/uc_crypto.c

```diff
@@ -0,0 +1,31 @@
+/*
+ * uc_crypto.c
+ *
+ * 	Tests underconstrained symbolic execution with
+ * 	Manticore in order to reproduce previous carry 
+ * 	propagation bug in TweetNaCl as well as to verify
+ * 	correctness in the current implementation.
+ *
+ *  	Compile with:
+ *  	  $ gcc -m32 -static -I. uc_crypto.c -o uc_crypto
+*/
+
+#include "vuln_tweetnacl.h"
+#include <unistd.h>
+#include <stdlib.h>
+
+/* Since we can't efficiently analyze static/shared libraries, we can 
+ * instead make an empty entry point, but use Manticore to hook and jump
+ * to different functions.
+ */
+int main() 
+{
+	unsigned char n[32], r[32];
+
+	size_t len;
```

Comment:
```suggestion
	ssize_t len;
```
Otherwise, the `len<0` check later on is an impossible path.